File: VB/WindowsFormsApplication3/InplaceDataNavigator/InplaceDataNavigatorHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraBars

Namespace WindowsFormsApplication3
	Public Class InplaceDataNavigatorHelper
		Private privateowner As BarEditItem
		Private Property owner() As BarEditItem
			Get
				Return privateowner
			End Get
			Set(ByVal value As BarEditItem)
				privateowner = value
			End Set
		End Property
		Private privaterepo As RepositoryItemInplaceDataNavigator
		Private Property repo() As RepositoryItemInplaceDataNavigator
			Get
				Return privaterepo
			End Get
			Set(ByVal value As RepositoryItemInplaceDataNavigator)
				privaterepo = value
			End Set
		End Property

		Public Sub New(ByVal barItem As BarEditItem)
			owner = barItem

			repo = TryCast(owner.Edit, RepositoryItemInplaceDataNavigator)
			AddHandler repo.EditNavigator.PositionChanged, AddressOf EditNavigator_PositionChanged
		End Sub
		Protected Overrides Sub Finalize()
			RemoveHandler repo.EditNavigator.PositionChanged, AddressOf EditNavigator_PositionChanged
			owner = Nothing
			repo = Nothing
		End Sub

		Private Sub EditNavigator_PositionChanged(ByVal sender As Object, ByVal e As EventArgs)
			owner.Refresh()
		End Sub
	End Class
End Namespace

File: CS/WindowsFormsApplication3/InplaceDataNavigator/InplaceDataNavigatorHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraBars;

namespace WindowsFormsApplication3
{
    public class InplaceDataNavigatorHelper
    {
        BarEditItem owner { get; set; } //Change owner type if you want to place DataNavigator within another container
        RepositoryItemInplaceDataNavigator repo { get; set; }

        public InplaceDataNavigatorHelper(BarEditItem barItem)
        {
            owner = barItem;

            repo = owner.Edit as RepositoryItemInplaceDataNavigator;
            repo.EditNavigator.PositionChanged += EditNavigator_PositionChanged;
        }
        ~InplaceDataNavigatorHelper()
        {
            repo.EditNavigator.PositionChanged -= EditNavigator_PositionChanged;
            owner = null;
            repo = null;
        }

        void EditNavigator_PositionChanged(object sender, EventArgs e)
        {
            owner.Refresh();
        }
    }
}
